sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -362,6 +401,68 @@ object GpuWindowExpression {
     case anythingElse =>
       throw new UnsupportedOperationException(s"Unsupported window frame expression $anythingElse")
   }
+
+  /**
+   * Create a Scalar from boundary value according to order by column type
+   * @param orderByType the type of order by column
+   * @param value boundary value
+   * @return Scalar holding boundary value
+   */
+  def createRangeWindowBoundary(orderByType: DType, value: Long): Scalar = {
+    orderByType match {
+      case DType.INT8 => Scalar.fromByte(value.toByte)
+      case DType.INT16 => Scalar.fromShort(value.toShort)
+      case DType.INT32 => Scalar.fromInt(value.toInt)
+      case DType.INT64 => Scalar.fromLong(value)
+      case DType.TIMESTAMP_DAYS => Scalar.durationFromLong(DType.DURATION_DAYS, value)
+      case DType.TIMESTAMP_SECONDS => Scalar.durationFromLong(DType.DURATION_SECONDS, value)
+      case DType.TIMESTAMP_MILLISECONDS =>
+        Scalar.durationFromLong(DType.DURATION_MILLISECONDS, value)
+      case DType.TIMESTAMP_MICROSECONDS =>
+        Scalar.durationFromLong(DType.DURATION_MICROSECONDS, value)
+      case DType.TIMESTAMP_NANOSECONDS => Scalar.durationFromLong(DType.DURATION_NANOSECONDS, value)
+      case _ => Scalar.fromNull(orderByType)
```

Comment:
Do we really want to ignore `value` here and assume the caller wants a null Scalar here, or is throwing an exception more appropriate?

Reply:
Done